application/cms/forms.py

```diff
@@ -425,3 +428,21 @@ def __init__(self, data_sources: Sequence[DataSource], *args, **kwargs):
 
         hints = {data_source.id: self._build_data_source_hint(data_source) for data_source in data_sources}
         self.data_sources.choices_hints = hints
+
+
+class SelectOrCreateDataSourceForm(SelectMultipleDataSourcesForm):
+    search_query = HiddenField()  # Retain the original query for the redirect flow when form was submitted with errors
+    data_sources = RDURadioField(
```

Comment:
Rename to `data_source_id` (singular) to make it clear that you're only posting a single ID (rather than an array of IDs, as in the previous checkbox form)?

Reply:
I'll update it to be singular.